packages/components/pages/f-account-info/src/components/AccountInfo.vue

```diff
@@ -9,61 +9,104 @@
             {{ $t('yourDetails') }}
         </h2>
 
+        <email-address-field :email-address="values.emailAddress" />
+
         <form
             method="post"
             @submit.prevent="onFormSubmit">
             <form-field
-                :label-text="$t('fields.emailAddressLabel')"
-                disabled
-                :placeholder="$t('fields.emailAddressPlaceholder')"
-                class="u-spacingBottom--large"
-                :value="fields.emailAddress" />
-
-            <f-link
-                is-distinct
-                :class="$style['c-accountInfo-customerCareLink']"
-                href="/help"
-                target="_blank">
-                {{ $t('contactCustomerCareTeam') }}
-            </f-link>
-
-            <form-field
+                v-model="fields.firstName"
+                maxlength="50"
                 :label-text="$t('fields.firstNameLabel')"
                 :placeholder="$t('fields.firstNamePlaceholder')"
-                :value="fields.firstName" />
+                @blur="onBlur('firstName')">
+                <template
+                    v-if="$v.fields.firstName.$invalid"
+                    #error>
+                    <f-error-message v-show="!$v.fields.firstName.required && $v.fields.firstName.$dirty">
+                        {{ $t('validation.firstNameRequired') }}
+                    </f-error-message>
+                    <f-error-message v-show="!$v.fields.firstName.isValidName">
+                        {{ $t('validation.firstNameInvalid') }}
+                    </f-error-message>
+                </template>
+            </form-field>
 
             <form-field
+                v-model="fields.lastName"
+                maxlength="50"
                 :label-text="$t('fields.lastNameLabel')"
                 :placeholder="$t('fields.lastNamePlaceholder')"
-                :value="fields.lastName" />
+                @blur="onBlur('lastName')">
+                <template
+                    v-if="$v.fields.lastName.$invalid"
+                    #error>
+                    <f-error-message v-show="!$v.fields.lastName.required && $v.fields.lastName.$dirty">
+                        {{ $t('validation.lastNameRequired') }}
+                    </f-error-message>
+                    <f-error-message v-show="!$v.fields.lastName.isValidName">
+                        {{ $t('validation.lastNameInvalid') }}
+                    </f-error-message>
+                </template>
+            </form-field>
 
             <h2
                 class="u-spacingBottom--large">
                 {{ $t('deliveryAddress') }}
             </h2>
 
             <form-field
+                v-model="fields.line1"
```

Comment:
Do we need to protect these address lines with a max length?

Reply:
They weren't in the checkout as far as I could see; I will double check